scopes/lanes/lanes/lanes.ui.runtime.tsx

```diff
@@ -0,0 +1,156 @@
+    Slot.withType<NavigationSlot>(),
+  ];
+
+  constructor(
+    private uiUi: UiUI,
+    private componentUi: ComponentUI,
+    private routeSlot: RouteSlot,
+    private menuRouteSlot: RouteSlot,
+    private navSlot: OrderedNavigationSlot,
+    private menuItemSlot: MenuItemSlot,
+    private reactRouter: ReactRouterUI,
+    private workspace?: WorkspaceUI,
+    private scope?: ScopeUI
+  ) {
+    this.hostAspect = workspace || scope;
+    this.lanesHost = workspace ? 'workspace' : 'scope';
+    this.host = workspace ? WorkspaceAspect.id : ScopeAspect.id;
+  }
+
+  private readonly lanesHost: LanesHost;
+  private readonly hostAspect?: WorkspaceUI | ScopeUI;
+  private readonly host: string;
+
+  private registerExplicitHostRoutes() {
+    if (this.hostAspect) {
+      this.hostAspect.registerRoutes([
+        {
+          path: laneComponentUrlRegex,
+          children: this.componentUi.getComponentUI(this.host),
+        },
+        {
+          path: laneRouteUrlRegex,
+          children: <LanesOverview routeSlot={this.routeSlot} />,
+        },
+      ]);
+      this.hostAspect.registerMenuRoutes([
+        {
+          path: laneComponentUrlRegex,
+          children: this.componentUi.getMenu(this.host),
+        },
+        {
+          path: laneRouteUrlRegex,
+          children: <LanesOverviewMenu navigationSlot={this.navSlot} host={this.host} />,
+        },
+      ]);
+    }
+  }
+
+  private registerExplicitLanesRoutes() {
+    this.registerNavigation({
+      href: '~gallery',
+      children: 'Gallery',
+    });
+    this.registerNavigation({
+      href: '',
+      children: 'Gallery',
+    });
+  }
+
+  private registerExplicitRoutes() {
+    this.registerExplicitHostRoutes();
+    this.registerExplicitLanesRoutes();
+  }
+
+  private renderContext = ({ children }: { children: ReactNode }) => {
+    return LanesProvider({ host: this.lanesHost, reactRouter: this.reactRouter, children });
+  };
+
+  registerRoute(route: RouteProps) {
+    this.routeSlot.register(route);
+    return this;
+  }
+
+  registerNavigation(nav: NavLinkProps, order?: number) {
+    this.navSlot.register({
+      props: nav,
+      order,
+    });
+  }
+
+  registerDrawers(...drawers: DrawerType[]) {
+    if (this.hostAspect) {
+      this.hostAspect.registerDrawers(...drawers);
+    }
+    return this;
+  }
+
+  static async provider(
+    [uiUi, reactRouter, componentUi]: [UiUI, ReactRouterUI, ComponentUI],
+    _,
+    [routeSlot, menuItemSlot, menuRouteSlot, navSlot]: [RouteSlot, MenuItemSlot, RouteSlot, OrderedNavigationSlot],
+    harmony: Harmony
+  ) {
+    const { config } = harmony;
+    const host = String(config.get('teambit.harmony/bit'));
+    let workspace: WorkspaceUI | undefined;
+    let scope: ScopeUI | undefined;
+    if (host === WorkspaceAspect.id) {
+      workspace = harmony.get<WorkspaceUI>(WorkspaceAspect.id);
+    }
+    if (host === ScopeAspect.id) {
+      scope = harmony.get<ScopeUI>(ScopeAspect.id);
+    }
```

Comment:
maybe we need a "host" aspect to provide the relevant workspace/scope, it would solve this age old problem. consult with David / Gilad

Reply:
discussed this with Ran, having a host aspect doesn't work because it ties us to a common interface. They essentially are different entities with different behaviour. That's why you will notice in `create-root` we pass the hostExtensionName via `harmony.config` and try to inject them dynamically instead of the provider.

Reply:
ok